monkey/infection_monkey/propagation_credentials_repository/propagation_credentials_repository.py

```diff
@@ -0,0 +1,53 @@
+import logging
+import time
+from itertools import chain
+from multiprocessing import get_context
+from typing import Iterable
+
+from common.credentials import Credentials
+from infection_monkey.i_control_channel import IControlChannel
+
+from .i_propagation_credentials_repository import IPropagationCredentialsRepository
+
+logger = logging.getLogger(__name__)
+
+CREDENTIALS_POLL_PERIOD_SEC = 10
+
+
+class PropagationCredentialsRepository(IPropagationCredentialsRepository):
+    """
+    Repository that stores credentials on the island and saves/gets credentials by using
+    command and control channel
+    """
+
+    def __init__(
+        self, control_channel: IControlChannel, polling_period: float = CREDENTIALS_POLL_PERIOD_SEC
+    ):
+        self._control_channel = control_channel
+        self._polling_period = polling_period
+        context = get_context("spawn")
+        self._next_update_time = context.Value("d", 0)
+        self._stored_credentials = context.Manager().list()
+
+    def add_credentials(self, credentials_to_add: Iterable[Credentials]):
+        logger.debug("Adding credentials")
+        self._stored_credentials.extend(set(chain(self._stored_credentials, credentials_to_add)))
+
+    def get_credentials(self) -> Iterable[Credentials]:
+        try:
+            with self._next_update_time.get_lock():
```

Comment:
This may be a bit confusing. Perhaps it would be better to just create `self._lock` and use that to syncronize?